model/internal/cmd/pdatagen/internal/base_fields.go

```diff
@@ -91,7 +91,14 @@ const copyToValueOneOfMessageTemplate = `	case ${typeName}:
 		ms.${fieldName}().CopyTo(dest.${fieldName}())`
 
 const accessorsOneOfPrimitiveTemplate = `// ${fieldName} returns the ${lowerFieldName} associated with this ${structName}.
+// Calling this function when ValueType() != ${valueType} will cause a panic.
 func (ms ${structName}) ${fieldName}() ${returnType} {
+	return (*ms.orig).Value.(*${originStructType}).${originFieldName}
+}
+
+// ${deprecatedFieldName} returns the ${lowerFieldName} associated with this ${structName}.
+// Deprecated: [v0.47.0] Use ${fieldName} instead.
+func (ms ${structName}) ${deprecatedFieldName}() ${returnType} {
 	return (*ms.orig).Get${originFieldName}()
 }
 
```

Comment:
I don't think we should generate this via the generator. You can write these manually.